hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java

```diff
@@ -449,6 +459,10 @@ private boolean createConfigFiles(String confDirPath, String kernelsDir, String
       sparkMagicParams.put(Settings.SPARK_EXECUTOR_EXTRACLASSPATH, new ConfigProperty(
           "spark_executor_extraClassPath", HopsUtils.APPEND_PATH, extraClassPath));
 
+      sparkMagicParams.put(Settings.SPARK_PACKAGES, new ConfigProperty(
```

Comment:
@Limmen @robzor92 @SirOibaf I suggest we add this avro dependency in spark-chef to the default spark jars. same way we do for parquet. If Hopsworks does not have internet access it won't be able to fetch the dependency from the maven central repo. If someone want to use different versions of these dependencies, they can add them as local resources or `spark.yarn.dist.jars`.

@Limmen Also, the property `spark_jars_packages` should not be ignore as users might provide additional dependencies.

Reply:
@tkakantousis It is not ignored. 
`
      //Prepare spark maven packages
      StringBuilder sparkPackages = new StringBuilder();
      //avro data source used in Feature Store
      sparkPackages.append(settings.getSparkAvroPackageName());
      //user-defined packages
      if(userSparkProperties.containsKey(Settings.SPARK_PACKAGES)){
        sparkPackages.append(",");
        sparkPackages.append(userSparkProperties.get(Settings.SPARK_PACKAGES));
      }
`
I've tested it on VM, you can add your own packages fine.

Reply:
@Limmen yes you are right, I mixed up the HopsUtils properties.

Reply:
@Limmen we don't need the localresources for this PR. LocalResources is meant for dependencies that are in HopsFs and the user wants to have them distributed to the executors. I think we can close this PR. We can the avro dependency in spark-chef ([link](https://github.com/logicalclocks/spark-chef/blob/3a962d079a2b600778e228616285a6a7a32cf095/recipes/install.rb#L74))together with the parquet dependencies. Then spark-avro will be in the classpath of the executors.